pkg/server/api/bundle/v1/service.go

```diff
@@ -186,49 +248,36 @@ func (s *Service) BatchDeleteFederatedBundle(ctx context.Context, req *bundle.Ba
 	return nil, status.Errorf(codes.Unimplemented, "method BatchDeleteFederatedBundle not implemented")
 }
 
-func applyMask(b *common.Bundle, mask *types.BundleMask) (*types.Bundle, error) {
+func applyBundleMask(b *types.Bundle, mask *types.BundleMask) {
 	if mask == nil {
-		mask = defaultMask
+		return
 	}
 
-	out := &types.Bundle{}
-	if mask.TrustDomain {
-		td, err := spiffeid.TrustDomainFromString(b.TrustDomainId)
-		if err != nil {
-			return nil, err
-		}
-		out.TrustDomain = td.String()
+	if !mask.RefreshHint {
+		b.RefreshHint = 0
 	}
 
-	if mask.RefreshHint {
-		out.RefreshHint = b.RefreshHint
+	if !mask.SequenceNumber {
+		b.SequenceNumber = 0
 	}
 
-	if mask.SequenceNumber {
-		out.SequenceNumber = 0
+	if !mask.TrustDomain {
+		b.TrustDomain = ""
 	}
 
-	if mask.X509Authorities {
-		var authorities []*types.X509Certificate
-		for _, rootCA := range b.RootCas {
-			authorities = append(authorities, &types.X509Certificate{
-				Asn1: rootCA.DerBytes,
-			})
-		}
-		out.X509Authorities = authorities
+	if !mask.X509Authorities {
+		b.X509Authorities = nil
 	}
 
-	if mask.JwtAuthorities {
-		var authorities []*types.JWTKey
-		for _, JWTSigningKey := range b.JwtSigningKeys {
-			authorities = append(authorities, &types.JWTKey{
-				PublicKey: JWTSigningKey.PkixBytes,
-				KeyId:     JWTSigningKey.Kid,
-				ExpiresAt: JWTSigningKey.NotAfter,
-			})
-		}
-		out.JwtAuthorities = authorities
+	if !mask.JwtAuthorities {
+		b.JwtAuthorities = nil
 	}
+}
 
-	return out, nil
+// CreateStatus creates a proto Status
+func CreateStatus(code codes.Code, format string, a ...interface{}) *types.Status {
```

Comment:
I dont think here is a good place for this method you have status methods on api: https://github.com/spiffe/spire/blob/master/pkg/server/api/status.go#L12